datafusion-examples/examples/deserialize_to_struct.rs

```diff
@@ -15,61 +15,61 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use arrow::array::AsArray;
+use arrow::datatypes::{Float64Type, Int32Type};
 use datafusion::error::Result;
 use datafusion::prelude::*;
-use serde::Deserialize;
+use futures::StreamExt;
 
 /// This example shows that it is possible to convert query results into Rust structs .
-/// It will collect the query results into RecordBatch, then convert it to serde_json::Value.
```

Comment:
https://github.com/apache/arrow-rs/pull/5318 deprecated the serde_json based APIs